### PR TITLE
DG-16983 - add Dockerfile

### PR DESCRIPTION
**Pull request by mDantas**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+RUN mkdir -p /go/src/github.com/lyft/ratelimit/
+WORKDIR /go/src/github.com/lyft/ratelimit/
+COPY . .
+
+# ratelimit startup
+RUN export REDIS_SOCKET_TYPE=tcp
+RUN export REDIS_URL=localhost:6379
+RUN make bootstrap
+RUN make compile
+RUN USE_STATSD=false LOG_LEVEL=debug REDIS_SOCKET_TYPE=tcp REDIS_URL=localhost:6379 RUNTIME_ROOT=/home/user/src/runtime/data RUNTIME_SUBDIRECTORY=ratelimit
```
Comment (junr03) [inline Dockerfile]: @mDantas I am not sure what you are trying to do here? Why do you export lines 9 and 10, if they are been reset here. Also do you intend to start the ratelimit service in this line?

**Comment by mfowlewebs** [comment]
I have pushed a more minimal alternative. As opposed to using a Go based container and building Ratelimit in the container, this build copies a local build into an alpine base image. There is a Makefile target to run this build, using the git sha as the tag.

File: Makefile [hunk 2]
```diff
@@ -45,3 +45,7 @@ tests: compile
 .PHONY: proto
 proto:
 	script/generate_proto
+
+.PHONY: docker
+docker:
```
Comment (junr03) [inline Makefile]: This target should probably have a `tests` dependency to make sure you have a binary (that passes tests) to copy into the docker container.

**Comment by junr03** [comment]
@mDantas @mfowlewebs can you both please sign the CLA https://github.com/lyft/ratelimit/blob/master/CONTRIBUTING.md#cla

**Comment by junr03** [comment]
@mfowlewebs I am missing your signature in the CLA. I will close for now until you sign. Please let me know when you do.

**Comment by mfowlewebs** [comment]
@junr03 CLA signed. Thanks!

**Pull request merged**